libs/core/src/lib/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,268 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    forwardRef,
+    HostBinding,
+    Input,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+
+const filesizemap = new Map([['KB', 1024], ['MB', 1048576], ['GB', 1073741824], ['TB', 1099511627776]]);
+/**
+ * Tool to facilitate the input of files from the user.
+ * It supports drag and drop, multiple input, max file size and more.
+ * The drag events make it very easy to create and style elements like a dropzone.
+ */
+@Component({
+    selector: 'fd-file-uploader',
+    templateUrl: './file-uploader.component.html',
+    styleUrls: ['./file-uploader.component.scss'],
+    host: {
+        '(blur)': 'onTouched()'
+    },
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => FileUploaderComponent),
+            multi: true
+        }
+    ],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FileUploaderComponent implements ControlValueAccessor {
+    /** @hidden */
+    @HostBinding('class.fd-file-uploader')
+    fdFileInputClass = true;
+
+    /** @hidden */
+    @ViewChild('fileInput')
+    inputRef: ElementRef;
+
+    @ViewChild('textInput')
+    inputRefText: ElementRef;
+
+    /** Whether the file input is disabled. */
+    @Input()
+    disabled = false;
+
+    /** Whether the file input should accept multiple files. */
+    @Input()
+    multiple = true;
+
+    /** Accepted file extensions. Format: `'.png,.jpg'`. */
+    @Input()
+    accept: string;
+
+    /** Whether the file input accepts drag and dropped files. */
+    @Input()
+    dragndrop = true;
+
+    /** Max file size in bytes that the input will accept. */
+    @Input()
+    maxFileSize: string;
```

Comment:
In my opinion it will be a lot cleaner both in our code and clearer from the app developer's perspective to use type number here and stick with just bytes

Reply:
This was discussed during design meeting, And decided to have size with human readable format like KB, MB, GB, TB.